+neuron/Segment.m

```diff
@@ -32,18 +32,34 @@ function push(self)
 
         function varargout = subsref(self, S)
         % Call a class method or dynamic property.
-            % Is the provided method listed above?
-            if ismethod(self, S(1).subs)
-                [varargout{1:nargout}] = builtin('subsref', self, S);
-            % Are we trying to directly access a class property?
-            elseif (isa(S(1).subs, "char") && length(S) == 1 && isprop(self, S(1).subs))
-                [varargout{1:nargout}] = self.(S(1).subs);
-            % If not a class property, return a Section range ref
-            elseif (isa(S(1).subs, "char") && length(S) == 1)
-                ref = self.ref(S(1).subs);
-                [varargout{1:nargout}] = ref.get();
+            if S(1).type == "."
+                % Are we trying to directly access a class property?
+                if isprop(self, S(1).subs)
+                    [varargout{1:nargout}] = self.(S(1).subs);
+                    n_processed = 1;  % Number of elements of S to process.
+                % Is the provided method listed above?
+                elseif numel(S) > 1 && ismethod(self, S(1).subs)
+                    [varargout{1:nargout}] = builtin('subsref', self, S(1:2));
+                    n_processed = 2;  % Number of elements of S to process.
+                % If not a class property, return a Section range ref
```

Comment:
Is this really the only other option? I'd kind of expect an error in the else, and some explicit check on input that matches a range ref request.